sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4245,42 +4230,42 @@ object UpdateOuterReferences extends Rule[LogicalPlan] {
 }
 
 /**
- * The rule `ResolveAggregationFunctions` in the main resolution batch creates
- * [[TempResolvedColumn]] in filter conditions and sort expressions to hold the temporarily resolved
- * column with `agg.child`. When filter conditions or sort expressions are resolved,
- * `ResolveAggregationFunctions` will replace [[TempResolvedColumn]], to [[AttributeReference]] if
- * it's inside aggregate functions or group expressions, or to [[UnresolvedAttribute]] otherwise,
- * hoping other rules can resolve it.
+ * The rule `ResolveReferences` in the main resolution batch creates [[TempResolvedColumn]] in
+ * UnresolvedHaving/Filter/Sort to hold the temporarily resolved column with `agg.child`.
  *
- * This rule runs after the main resolution batch, and can still hit [[TempResolvedColumn]] if
- * filter conditions or sort expressions are not resolved. When this happens, there is no point to
- * turn [[TempResolvedColumn]] to [[UnresolvedAttribute]], as we can't resolve the column
- * differently, and query will fail. This rule strips all [[TempResolvedColumn]]s in Filter/Sort and
- * turns them to [[AttributeReference]] so that the error message can tell users why the filter
- * conditions or sort expressions were not resolved.
+ * If the expression hosting [[TempResolvedColumn]] is fully resolved, the rule
+ * `ResolveAggregationFunctions` will replace [[TempResolvedColumn]] with [[AttributeReference]] if
+ * it's inside aggregate functions or group expressions, or mark it as `hasTried` otherwise, hoping
```

Comment:
I'm not sure I read this correctly. `hasTried` will be set to true, if the expression hosting `TempResolvedColumn` cannot be resolved, OR if it is not inside aggregate functions or group expressions?

Reply:
if it is not inside aggregate functions or group expressions. Let me rephase the doc a bit more.